core/src/main/java/discord4j/core/spec/AuditSpec.java

```diff
@@ -18,10 +18,22 @@
 
 import reactor.util.annotation.Nullable;
 
+/** A spec which can optionally have a reason in the audit logs when built. */
 public interface AuditSpec<T> extends Spec<T> {
 
+    /**
+     * Sets the audit log reason to show in the audit logs when the spec is built.
+     *
+     * @param reason The audit log reason.
+     * @return This spec.
+     */
     AuditSpec<T> setReason(@Nullable String reason);
 
+    /**
+     * Returns the current audit log reason set on the spec.
+     *
+     * @return The current audit log reason.
+     */
     @Nullable
     String getReason();
```

Comment:
This function like `Spec#asRequest` is not a truly supported method. Users should never call this, and that should be made clear.

Reply:
how do you want me to make it clear to users? aren't there annotations I can add to the method now? or I could notify them in the documentation or remove the docs for the method altogether.